### PR TITLE
Fix: On --init, print message when package.json is invalid (fixes #6257)

### LINKED ISSUE #6257: Syntax Error on eslint --init (ask questions about style) when package.json is malformed
On running `eslint --init` I get the following error. I've tried a few different choices and all result in the same problem. The init phase never succeeds.
```
$ ./node_modules/eslint/bin/eslint.js --init
? How would you like to configure ESLint? Answer questions about your style
? Are you using ECMAScript 6 features? Yes
? Are you using ES6 modules? Yes
? Where will your code run? Browser
? Do you use CommonJS? Yes
? Do you use JSX? Yes
? Do you use React Yes
? What style of indentation do you use? Spaces
? What quotes do you use for strings? Single
? What line endings do you use? Unix
? Do you require semicolons? Yes
? What format do you want your config file to be in? JavaScript
Unexpected token }
SyntaxError: Unexpected token }
    at Object.parse (native)
    at check (/home/peter/Projects/rikesh-accounting-web/node_modules/eslint/lib/util/npm-util.js:76:25)
    at Object.checkDevDeps (/home/peter/Projects/rikesh-accounting-web/node_modules/eslint/lib/util/npm-util.js:116:12)
    at installModules (/home/peter/Projects/rikesh-accounting-web/node_modules/eslint/lib/config/config-initializer.js:86:29)
    at processAnswers (/home/peter/Projects/rikesh-accounting-web/node_modules/eslint/lib/config/config-initializer.js:262:5)
    at null.completed (/home/peter/Projects/rikesh-accounting-web/node_modules/eslint/lib/config/config-initializer.js:469:30)
    at PromptUI.onCompletion (/home/peter/Projects/rikesh-accounting-web/node_modules/eslint/node_modules/inquirer/lib/ui/prompt.js:57:10)
    at AnonymousObserver.Rx.AnonymousObserver.AnonymousObserver.completed (/home/peter/Projects/rikesh-accounting-web/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:1550:12)
    at AnonymousObserver.Rx.internals.AbstractObserver.AbstractObserver.onCompleted (/home/peter/Projects/rikesh-accounting-web/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:1489:14)
    at Subject.Rx.Subject.addProperties.onCompleted (/home/peter/Projects/rikesh-accounting-web/node_modules/eslint/node_modules/inquirer/node_modules/rx-lite/rx.lite.js:5871:19)
```

**What version of ESLint are you using?**
```
$ ./node_modules/eslint/bin/eslint.js -v
v2.10.2
```
```
$ node -v
v4.4.5
```

**What parser (default, Babel-ESLint, etc.) are you using?**

n/a

**Please show your full configuration:**

n/a

**What did you do? Please include the actual source code causing the issue.**

See above.

**What did you expect to happen?**

To successfully create the .eslintrc

**What actually happened? Please include the actual, raw output from ESLint.**

Syntax Error

### PR DESCRIPTION
**Pull request by kaicataldo**

### CODE REVIEW

File: tests/lib/util/npm-util.js [hunk 1]
```diff
@@ -66,6 +67,27 @@ describe("npmUtil", function() {
             fs.existsSync.restore();
             fs.readFileSync.restore();
         });
+
+        it("should throw with message when parsing invalid package.json", function() {
+            var logInfo = sinon.stub(log, "info");
```
Comment (platinumazure) [inline tests/lib/util/npm-util.js]: If we use `sinon.sandbox.create()` in test setup and `sandbox.restore()` in test teardown, then we don't need to worry about restoring at the end of a test (and as a bonus it is exception safe as well). Just a suggestion :smile:

Comment (nzakas) [inline tests/lib/util/npm-util.js]: Yeah, we should do that to make the test resilient.

Comment (kaicataldo) [inline tests/lib/util/npm-util.js]: Sure! Not super experienced with `sinon` - would this happen in a `beforeEach`/`afterEach` in the outermost `describe`?

Comment (platinumazure) [inline tests/lib/util/npm-util.js]: @kaicataldo Yes, that's a good place for it. Something like this:

```js
describe("something", function () {
    var sandbox;

    beforeEach(function () {
        sandbox = sinon.sandbox.create();
    });

    afterEach(function () {
        sandbox.restore();
    });

    it("should do something", function () {
        var logInfo = sandbox.stub(logger, "info");
        // etc.
    });
```

Comment (platinumazure) [inline tests/lib/util/npm-util.js]: Docs: http://sinonjs.org/docs/#sandbox

Comment (nzakas) [inline tests/lib/util/npm-util.js]: Example: https://github.com/eslint/eslint/blob/master/tests/lib/eslint.js#L78

**Comment by nzakas** [comment]
I'm just going to update the tests so I can include this in the release.

**Comment by ilyavolodin** [comment]
LGTM

**Comment by nzakas** [comment]
@kaicataldo leaving this branch for you to delete (because you created it -- don't want to hijack your branch).

**Comment by kaicataldo** [comment]
Thanks for updating the tests @nzakas

**Pull request merged**